### PR TITLE
Remove Unused ChatChannel Indexing

### PR DESCRIPTION
**Pull request by mstruve**
## What type of PR is this? (check all applicable)
- [x] Refactor

## Description
We are indexing chat channels but we are never searching for them as far as I can tell. I attempted to confirm this by doing a search of the app for any indication of searching that index. I found none. We only search on the `SecuredChatChannelMembership` index. I then went into Algolia and found that no recent searches had been conducted, only updating documents. 
 
## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/projects/6#card-32289313

## Added to documentation?
- [x] no documentation needed

alt_text

### CODE REVIEW

File: app/models/chat_channel.rb [hunk 1]
```diff
@@ -19,18 +18,6 @@ class ChatChannel < ApplicationRecord
...
   validates :slug, uniqueness: true, presence: true
 
-  algoliasearch index_name: "SecuredChatChannel_#{Rails.env}" do
-    attribute :id, :viewable_by, :slug, :channel_type,
-              :channel_name, :last_message_at, :status,
-              :messages_count, :channel_human_names, :channel_mod_ids, :pending_users_select_fields,
-              :description
-    searchableAttributes %i[channel_name channel_slug channel_human_names]
-    attributesForFaceting ["filterOnly(viewable_by)", "filterOnly(status)", "filterOnly(channel_type)"]
-    ranking ["desc(last_message_at)"]
```
Comment (rhymes) [inline app/models/chat_channel.rb]: some of these attributes are methods just used by the index that can also be removed: 

- `viewable_by`
- `messages_count`

**Comment by lightalloy** [comment]
Looks good to me :rocket:

**Comment by benhalpern** [comment]
Yes, I believe this is correct. I changed things on the client from reading from the `ChatChannels` index to reading from `ChatChannelMemberships` a while back.

**Comment by mstruve** [comment]
Excellent! Can I get some actual Approvals instead of comments so I can merge? Please! :) 

@sarthology if you plan on searching ChatChannels then lets get that setup in Elasticsearch right from the start! Then it's one less thing we have to move over.

**Comment by sarthology** [comment]
@rhymes @mstruve Yup, Not using ChatChannels index or planning to use it. It's Good to go. 👍🏽

**Pull request merged**